plugin/auth_pam/auth_pam.c

```diff
@@ -19,6 +19,9 @@
 #include <mysql/plugin_auth.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <errno.h>
 #include <security/pam_appl.h>
```

Comment:
Are these three files guaranteed to exist when pam-devel headers are?

Reply:
yes. tested on solaris, osx, freebsd and linux. There are also the ones needed for the syscalls performed, which come from POSIX standards